### ISSUE TITLE
UI/UX

### ISSUE THREAD

**Issue by ellogwen** [jarneson/godot-ply#62]
This issue is for discussing the upcoming UI/UX changes.

The following bullet list are just the general topic. Feel free to add your wishes

- Shortcuts
- Radial Menu
- Context aware actions -> Move currently needed actions in the godot spatial top bar instead of/in addition to one large, vertical menu
- Grid Layout -> Relayout the current menu panel to feature a more compact, but more verbose layout. A grid instead of a list may be suitable
- Color coating -> I will introduce primary, secondary, accent and other colours that will get used within.
- Instant feedback. Currently the mesh get only regenerated when you commit the handle action. In my plugin I got instant generating to work, I'll take a look if this can be done as well
- Generator Preview - I will add a preview to the generator window or i will generalize the mesh generator process to use the same ui/ux for all (upcoming) generators

**Comment by GeorgeS2019** [comment]
@ellogwen 
I respect the ambition here. It could help for a new user there to see where is the current status in terms of the different modes of blender that are now capable through this AddOn.  

This gives the new users what is the status and where they could contribute.

### Blender [Modelling Modes](https://docs.blender.org/manual/en/latest/modeling/introduction.html#modes)
- [ ] Meshes
- [ ] Curves
- [ ] Surfaces
- [ ] Metaballs
- [ ] Text objects
- [ ] Lattice

**Comment by fire** [comment]
Snapping has an implmentation here @ellogwen .https://github.com/jgillich/godot-snappy

**Comment by jarneson** [comment]
@GeorgeS2019 I believe such a list would be better served as individual issues. This issue is largely about broader "everywhere" kind of UX. Although perhaps this issue should just be separated into many.

**Comment by Frontrider** [comment]
I *think* this is the UX update this needs as a start. 
![image](https://user-images.githubusercontent.com/6755034/183653633-7ed37e4e-a55f-4506-ae2e-77e1d5ab4659.png)
A lot more compact and leaves more room to see what I'm doing. I originally tried to go for a bottom panel, but that is just overkill.

Some icons are questionable there, but you get the gist of it. Only thing I did not hook up right now, is the mesh generator.

**Comment by Frontrider** [comment]
The side panel needs serious reworks if it were to be used.

Also, the alternative to the side panel is to do what the tilemap editor does in 4.0 and migrate to the bottom.
![image](https://user-images.githubusercontent.com/6755034/183684147-39a46e0f-e82a-4233-88cb-ce6d792f5e78.png)

**Comment by Frontrider** [comment]
Bottom of the screen is already usually taken up by the tool for the thing I'm currently working with (animation, audio, console for debugging etc) any ways so it fits well.

**Comment by jarneson** [comment]
yeah, bottom panel could be good, in fact I was hoping to move the inspector plugin to the bottom so I can use it in distraction-free mode.

open to really anything as long as it hits my vague onboarding goal.

to reiterate: strong iconography would go a really long way to reducing my self-imposed constraints there, so if anyone can rope in someone to work on that would be HUGE.
